src/python/WMCore/MicroService/MSPileup/MSPileup.py

```diff
@@ -4,14 +4,125 @@
+from WMCore.MicroService.DataStructs.DefaultStructs import PILEUP_REPORT
+from WMCore.MicroService.MSPileup.MSPileupData import MSPileupData
+
+
+def mspileupError(doc):
+    """
+    Check MSPileup record for error
+    :return: None or raise cherrypy.HTTPError
+    """
+    if 'error' in doc:
+        msg = doc['message']
+        code = doc['code']
+        msg = f'MSPileupError: {msg}, code: {code}'
+        raise cherrypy.HTTPError(400, msg) from None
 
 
 class MSPileup(MSCore):
     """
     MSPileup provides whole logic behind the pileup WMCore module.
     """
 
-    def __init__(self, msConfig, logger=None, **kwargs):
-        super(MSPileup, self).__init__(msConfig, **kwargs)
+    def __init__(self, msConfig, **kwargs):
+        # TODO: so far we are testing this service w/o Rucio access
+        super(MSPileup, self).__init__(msConfig, skipRucio=True)
+        # TODO: if more generic approach will be required we'll need to switch to
+#         super(MSPileup, self).__init__(msConfig, **kwargs)
+
+        # for authz we should provide relevant section in MS configuration
+        # if no role/group is provided then authz_match used in HTTP APIs
+        # will always return valid state. Therefore, for test purposes we can
+        # use empty role and group, while for production setup we should set thenm up
+        authzDefaults = msConfig.get('authz_defaults', {})
+        self.role = authzDefaults.get('role', [])
+        self.group = authzDefaults.get('group', [])
+        self.mgr = MSPileupData(msConfig)
+
+    def status(self):
+        """
+        Provide MSPileup status API. We should extend it to check DB connection, etc.
+
+        :return: status dictionary
+        """
+        summary = dict(PILEUP_REPORT)
+        summary.update({'thread_id': current_thread().name})
+        return summary
+
+    def get(self, **kwargs):
+        """
+        Provide MSPileup get API
+
+        :param **kwargs: provide key=value (or spec) input
+        :return: results of MSPileup data layer (list of dicts)
+        """
+        spec = {}  # (get all full docs) retrieve a list with the full documentation of all the pileups
+        if 'spec' in kwargs:
+            # use specific spec (JSON query)
+            spec = kwargs['spec']
+        elif 'pileupName' in kwargs:
+            # get docs for given pileupName
+            spec = {'pileupName': kwargs['pileupName']}
+        elif 'campaign' in kwargs:
+            # get docs for given campaign
+            spec = {'campaign': kwargs['campaign']}
+        else:
+            for key, val in kwargs.items():
+                spec[key] = val
+
+        # check if filters are present and use it as projection fields
+        projection = {}
+        for key in kwargs.get('filters', []):
+            projection[key] = 1
+        results = self.mgr.getPileup(spec, projection)
+
+        return results
+
+    def query(self, spec, projection=None):
```

Comment:
How about renaming it to `queryDatabase`? Docstring can be improved as well.

Reply:
done